### PR TITLE
fix(suite): Apply filtering on the export of the transactions

### LINKED ISSUE #4723: Export filtered transactions
Request for a filtered export of transactions (by date, label, ... ) This feature works in Trezor Wallet.

**How to reproduce**
1. Go to Accounts
2. Add few labels
3. Filter according your label or by date (ie `>2021-11-30 & <2022-01-01`)
4. Export transactions

**Expected behaviour**
Export filtered transactions only (Like Trezor Wallet does)

**Actual behaviour**
Export all the transactions

## Original request from customer:
```
Hello, we use your Wallet in our business using transaction labeling (with invoice number) and monthly exports (for accounting) but none is currently working in your Suite.
I have enabled labeling, but cannot find a way to label a transaction. Transaction import does not show option how to add label.
Filter is working, I can filter and see monthly transactions, but then export (both PDF,CSV) still exports the whole history, not only what is filtered out.
Can this be improved before turning off the “Wallet”?
```

### PR DESCRIPTION
**Pull request by peter-sanderson**
## Description

Adds a filter to the export, so transaction that are exported are aligned with the filtered list.

## Related Issue 

Resolve https://github.com/trezor/trezor-suite/issues/4723

## Screenshots:
image
image

### CODE REVIEW

File: suite-common/wallet-core/src/transactions/transactionsThunks.ts [hunk 1]
```diff
@@ -223,10 +225,14 @@ export const exportTransactionsThunk = createThunk(
             account,
             accountName,
             type,
+            search,
```
Comment (dahaca) [inline suite-common/wallet-core/src/transactions/transactionsThunks.ts]:
```suggestion
            searchQuery,
```

File: packages/suite/src/views/wallet/transactions/TransactionList/TransactionListActions/ExportAction.tsx [hunk 2]
```diff
@@ -12,12 +12,15 @@ import { Account } from 'src/types/wallet';
 import { isFeatureFlagEnabled, getTxsPerPage } from '@suite-common/suite-utils';
 import { getTitleForNetwork, getTitleForCoinjoinAccount } from '@suite-common/wallet-utils';
 import { selectLabelingDataForSelectedAccount } from 'src/reducers/suite/metadataReducer';
+import { AccountLabels } from '@suite-common/metadata-types';
 
 export interface ExportActionProps {
     account: Account;
+    search: string;
```
Comment (dahaca) [inline packages/suite/src/views/wallet/transactions/TransactionList/TransactionListActions/ExportAction.tsx]:
```suggestion
    searchQuery: string;
```

File: suite-common/wallet-core/src/transactions/transactionsThunks.ts [hunk 3]
```diff
@@ -263,12 +269,18 @@ export const exportTransactionsThunk = createThunk(
                 })),
             }));
 
+        const filteredTransaction = advancedSearchTransactions(
```
Comment (dahaca) [inline suite-common/wallet-core/src/transactions/transactionsThunks.ts]:
```suggestion
        const filteredTransactions = advancedSearchTransactions(
```

Comment (dahaca) [reply suite-common/wallet-core/src/transactions/transactionsThunks.ts]: only filter when `search` is present?

File: packages/suite/src/views/wallet/transactions/TransactionList/TransactionListActions/ExportAction.tsx (hunk 2 above)
Comment (peter-sanderson) [reply packages/suite/src/views/wallet/transactions/TransactionList/TransactionListActions/ExportAction.tsx]: :heavy_check_mark:

File: suite-common/wallet-core/src/transactions/transactionsThunks.ts (hunk 3 above)
Comment (peter-sanderson) [reply suite-common/wallet-core/src/transactions/transactionsThunks.ts]: :heavy_check_mark:

File: suite-common/wallet-core/src/transactions/transactionsThunks.ts (hunk 1 above)
Comment (peter-sanderson) [reply suite-common/wallet-core/src/transactions/transactionsThunks.ts]: :heavy_check_mark:

**Review by dahaca — APPROVED** [review]

**Pull request merged**